### PR TITLE
[ML] add new delete trained model aliases API

### PR DESCRIPTION
**Pull request by benwtrent**
In addition to creating and re-assigning model aliases, users should be able to delete existing and unused model aliases.

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: docs/reference/ml/df-analytics/apis/delete-trained-models-aliases.asciidoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
+[role="xpack"]
+[testenv="platinum"]
+[[delete-trained-models-aliases]]
+= Delete Trained Models Aliases API
+[subs="attributes"]
+++++
+<titleabbrev>Delete Trained Models Aliases</titleabbrev>
+++++
+
+Deletes a trained models alias.
```
Comment (davidkyle) [inline docs/reference/ml/df-analytics/apis/delete-trained-models-aliases.asciidoc:10]: Is it "model's" possessive or singular "Deletes a trained model alias"

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelAction.java [hunk 2]
```diff
@@ -142,7 +142,7 @@ public ClusterState execute(final ClusterState currentState) {
         });
     }
 
-    private Set<String> getReferencedModelKeys(IngestMetadata ingestMetadata) {
+    static Set<String> getReferencedModelKeys(IngestMetadata ingestMetadata, IngestService ingestService) {
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelAction.java:145]: Unused?

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelAction.java (hunk 2 above)
Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelAction.java:145]: it is used also in the transport delete trained model alias class to gather all the referenced models to make sure a model alias is not used by a pipeline

File: docs/reference/ml/df-analytics/apis/delete-trained-models-aliases.asciidoc (hunk 1 above)
Comment (benwtrent) [reply docs/reference/ml/df-analytics/apis/delete-trained-models-aliases.asciidoc:10]: definitely

**Pull request merged**